From 4fcd8bdcffc4a18badc01ff7e68f08af42e4ec9e Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Thu, 11 Feb 2016 19:25:04 -0500
Subject: [PATCH] Add MC Util class

To handle things needed in NMS such as distance calculations, stack trace (for debugging) etc.

Placed in NMS package to avoid extra diff of imports/full package refs
---
 src/main/java/net/minecraft/server/MCUtil.java | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/src/main/java/net/minecraft/server/MCUtil.java b/src/main/java/net/minecraft/server/MCUtil.java
index d3a256c..a7c2595 100644
--- a/src/main/java/net/minecraft/server/MCUtil.java
+++ b/src/main/java/net/minecraft/server/MCUtil.java
@@ -1,5 +1,6 @@
 package net.minecraft.server;
 
+import org.apache.commons.lang.exception.ExceptionUtils;
 import org.bukkit.Location;
 
 import javax.annotation.Nullable;
@@ -10,6 +11,24 @@ public class MCUtil {
 
     private MCUtil() {}
 
+    /**
+     * Quickly generate a stack trace for current location
+     *
+     * @return Stacktrace
+     */
+    public static String stack() {
+        return ExceptionUtils.getFullStackTrace(new Throwable());
+    }
+
+    /**
+     * Quickly generate a stack trace for current location with message
+     *
+     * @param str
+     * @return Stacktrace
+     */
+    public static String stack(String str) {
+        return ExceptionUtils.getFullStackTrace(new Throwable(str));
+    }
 
     /**
      * Builds a chat componenent from a string.
-- 
2.8.2

